sa/sa.go

```diff
@@ -491,40 +549,61 @@ func (ssa *SQLStorageAuthority) MarkCertificateRevoked(ctx context.Context, seri
 
 	tx, err := ssa.dbMap.Begin()
 	if err != nil {
-		return
+		return err
 	}
 
-	var status core.CertificateStatus
+	var statusObj interface{}
+	var fields string
+
+	if features.Enabled(features.CertStatusOptimizationsMigrated) {
+		statusObj = &certStatusModelv2{}
+		fields = CertificateStatusFieldsv2
+	} else {
+		statusObj = &certStatusModelv1{}
+		fields = CertificateStatusFields
+	}
 	err = tx.SelectOne(
-		&status,
-		fmt.Sprintf("SELECT %s FROM certificateStatus WHERE serial = ?", CertificateStatusFields),
-		serial,
-	)
+		statusObj,
+		fmt.Sprintf("SELECT %s FROM certificateStatus WHERE serial = ?", fields),
+		serial)
 	if err == sql.ErrNoRows {
-		return Rollback(tx, fmt.Errorf("No certificate status found for %s", serial))
+		err = fmt.Errorf("No certificate with serial %s", serial)
+		err = Rollback(tx, err)
+		return err
 	}
 	if err != nil {
-		return Rollback(tx, err)
+		err = Rollback(tx, err)
+		return err
 	}
 
+	var n int64
 	now := ssa.clk.Now()
-	status.Status = core.OCSPStatusRevoked
-	status.RevokedDate = now
-	status.RevokedReason = reasonCode
+	if features.Enabled(features.CertStatusOptimizationsMigrated) {
+		status := statusObj.(*certStatusModelv2)
+		status.Status = core.OCSPStatusRevoked
+		status.RevokedDate = now
+		status.RevokedReason = reasonCode
+
+		n, err = tx.Update(status)
```

Comment:
If you assign `status` back to `statusObj` after changing the fields you could move the `tx.Update` outside of this block and make the err check below slightly more obvious, not necessary though.